actors/eam/src/lib.rs

```diff
@@ -177,21 +192,56 @@ fn create_actor(
     Ok(Return::from_exec4(ret, new_addr))
 }
 
-fn resolve_caller(rt: &mut impl Runtime) -> Result<EthAddress, ActorError> {
-    let caller_id = rt.message().caller().id().unwrap();
-    Ok(match rt.lookup_delegated_address(caller_id).map(|a| *a.payload()) {
-        Some(Payload::Delegated(addr)) if addr.namespace() == EAM_ACTOR_ID => EthAddress(
+fn resolve_eth_address(rt: &mut impl Runtime, actor_id: ActorID) -> Result<EthAddress, ActorError> {
+    match rt.lookup_delegated_address(actor_id).map(|a| *a.payload()) {
+        Some(Payload::Delegated(addr)) if addr.namespace() == EAM_ACTOR_ID => Ok(EthAddress(
             addr.subaddress()
                 .try_into()
                 .context_code(ExitCode::USR_FORBIDDEN, "caller's eth address isn't valid")?,
-        ),
-        _ => {
-            let mut bytes = [0u8; 20];
-            bytes[0] = 0xff;
-            bytes[12..].copy_from_slice(&caller_id.to_be_bytes());
-            EthAddress(bytes)
+        )),
+        _ => Err(actor_error!(forbidden; "caller doesn't have an eth address")),
+    }
+}
+
+fn resolve_caller_external(rt: &mut impl Runtime) -> Result<(EthAddress, EthAddress), ActorError> {
```

Comment:
We might have to deal with multisigs a here?

Reply:
We don't currently allow them, we expect EthAccount or Account; we'll have to be explicit about those.

But yes, we might want to allow multisig ownership of a newly created contract, but that's a can of worms we can open later.

Reply:
CreateExternal needs to happen from an account. We could introduce an _additional_ "create native" (or something like that), but, IMO, that's not critical right now.

We're also trying to encourage users to switch over to EVM multisigs.